enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,188 @@
+
+Note: This process will work only on the clusters where the feature was released as part of the Insights operator and will not be backported to older versions.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure.
+
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
+The OCM API provides an endpoint that will serve the updated `pull-secret` definition when the migration window is opened. The endpoint will be defined in a new `ClusterTransfer` API - more details are in https://gitlab.cee.redhat.com/service/uhc-account-manager/-/merge_requests/2716.
+The migration window will be opened for certain period of time (e.g 5 days) so the Insights Operator can safely update the secret.
+
+When the migration window is closed, the OCM API returns another HTTP status code and the Insights operator does nothing in such case.
+
+### Monitoring
+
+The OCM API will return time information providing the remaining time until the end of the current migration window. There will be an alert triggered when a certain number of days (1-2) remain in the current migration window.
+
+There is also plan to create an alert about pending `ClusterTransfer` on the OCM side - see https://issues.redhat.com/browse/SDB-2506,
+
+### Configuration
+
+The OCM API endpoint will be configurable in the Insights operator via the `support` secret (this is curenlty the only way to configure the Insights operator).
+
+### Disconnected clusters
+
+This feature will not work on disconnected clusters because there is no corresponding token (in the `pull-secret`) to communicate with Red Hat. We should emphasize this in https://docs.openshift.com/container-platform/4.9/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html.
+
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify that the Insights Operator is able to download and update the `pull-secret` content
+
+### Graduation Criteria
+
+Thera are no graduation criteria. This feature is planned as a fast track feature.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+Removing this feature will require updating the Insights operator code base.
```

Comment:
This can also be N/A. The section is mainly for enhancements that involves removing a deprecated feature.

Reply:
OK. Removed. Thank you.